modules/onboard-azure-eventhub/variables.tf

```diff
@@ -0,0 +1,175 @@
+variable "azure_eventhub_admin_email" {
+  description = "The email address to notify about the asset."
+  type        = string
+}
+
+variable "azure_eventhub_audit_pull_enabled" {
+  description = "If true, sonargateway will collect the audit logs for this system if it can."
+  type        = bool
+  default     = null
+}
+
+variable "azure_eventhub_format" {
+  description = "The type of audit data being sent to the Event Hub. Possible values are: AzureSQL_Managed, Blob, Cosmos_Mongo, Cosmos_SQL, Data_Explorer, Databricks_Workspace, File, Mariadb, Mysql, Postgresql, Postgresql_Flexible, Queue, Sql, Synapse, Table. Defaults to Sql."
+  type        = string
+  default     = "Sql"
+  validation {
+    condition = contains(
+      [
+        "AzureSQL_Managed",
+        "Blob",
+        "Cosmos_Mongo",
+        "Cosmos_SQL",
+        "Data_Explorer",
+        "Databricks_Workspace",
+        "File",
+        "Mariadb",
+        "Mysql",
+        "Postgresql",
+        "Postgresql_Flexible",
+        "Queue",
+        "Sql",
+        "Synapse",
+        "Table"
+      ],
+      var.azure_eventhub_format
+    )
+    error_message = "Invalid format. Possible values are: AzureSQL_Managed, Blob, Cosmos_Mongo, Cosmos_SQL, Data_Explorer, Databricks_Workspace, File, Mariadb, Mysql, Postgresql, Postgresql_Flexible, Queue, Sql, Synapse, Table."
+  }
+}
+
+variable "azure_eventhub_gateway_id" {
+  description = "Unique identifier (UID) attached to the jSonar machine controlling the asset"
+  type        = string
+}
+
+variable "azure_eventhub_parent_asset_id" {
+  description = "The asset_id of the azure asset that is sending its audit logs to this AZURE EVENTHUB asset."
+  type        = string
+  default     = null
+}
+
+variable "azure_eventhub_region" {
+  description = "Azure region containing the Event Hub."
+  type        = string
+  default     = null
+}
+
+variable "eventhub_message_retention" {
```

Comment:
Do we want to add that note about the possible values from the provider doc here? I.e.
```
When using a dedicated Event Hubs cluster, maximum value of message_retention is 90 days. 
When using a shared parent EventHub Namespace, maximum value is 7 days; or 1 day when using a Basic SKU for the shared parent EventHub Namespace.
```
(Relevant also for `eventhub_partition_count`)

Reply:
Considering this module doesn't support a dedicated `Event Hubs cluster`, I think we're okay leaving that out. Definitely agree with the namespace addition though.